### ISSUE TITLE
Improve performance of HttpProbe.filterUrl()

### ISSUE THREAD

**Issue by seabaylea** [RuntimeTools/appmetrics#57]
Profiling the overhead of appmetrics with the AcmeAir benchmark, and visualizing the data in the Health Center client, shows that 2.47% of CPU is being spent executing `HttpProbe.filterUrl()` and the functions it calls:
![image](https://cloud.githubusercontent.com/assets/11974522/11142893/8eea3a8e-89e8-11e5-9f48-7f2c0b596c2e.png)

`HttpProbe.filterUrl()` itself is only using 0.021% (Self %) and the functions it calls account for the rest. The called methods shows that almost all of this time is spent executing `Url.parse()`.

Whilst `Url.parse()` can parse the full range of URL types and break it down into constituent parts, the use case here is much more limited - we're just taking the `req.url` value from the HTTP request (which don't contain hostname/port etc), and stripping off any query on the end.

This can be done much more cheaply using a custom implementation rather than using `Url.parse()`, ie. stripping the URL at the first instance of '#' or '?'.

**Comment by seabaylea** [comment]
Using a simple `indexOf()` / `substring()` implementation like the following:

```js
var parse = function (url) {
	['?','#'].forEach(function (separator)  {
		var index = url.indexOf(separator);
		if (index !== -1) url = url.substring(0, index);
	});
	return url;
};
```
Shows a 3x improvement over `Url.parse()` for a URL with containing a '?' query. This means that the change should result in a ~1.6% reduction in the performance overhead of appmetrics.

**Comment by seabaylea** [comment]
Resolved by PR #59

**Comment by seabaylea** [comment]
Latest data has the cost of `HttpProbe.filterUrl()` and the functions it calls down to 0.24%
